[32mINFO    [0m pytest_operator.plugin:plugin.py:903 Model status:

Model  Controller           Cloud/Region         Version  SLA          Timestamp
test   localhost-localhost  localhost/localhost  3.4.5    unsupported  01:42:04Z

App                  Version  Status   Scale  Charm                Channel      Rev  Exposed  Message
another-database              error        2  postgresql                          1  no       hook failed: "leader-elected"
database                      waiting    0/1  postgresql                          0  no       waiting for machine
postgresql-test-app           active       2  postgresql-test-app  latest/edge  301  no       

Unit                    Workload  Agent       Machine  Public address  Ports  Message
another-database/0      waiting   executing   3        10.212.8.37            agent initialising
another-database/1*     error     idle        4        10.212.8.183           hook failed: "leader-elected"
database/0              waiting   allocating  2        10.212.8.92            waiting for machine
postgresql-test-app/0*  active    idle        0        10.212.8.9             
postgresql-test-app/1   active    idle        1        10.212.8.177           

Machine  State    Address       Inst id        Base          AZ  Message
0        started  10.212.8.9    juju-31e3e4-0  ubuntu@22.04      Running
1        started  10.212.8.177  juju-31e3e4-1  ubuntu@22.04      Running
2        pending  10.212.8.92   juju-31e3e4-2  ubuntu@22.04      Running
3        started  10.212.8.37   juju-31e3e4-3  ubuntu@22.04      Running
4        started  10.212.8.183  juju-31e3e4-4  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:909 Juju error logs:

machine-4: 01:41:38 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:41:38 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-another-database-1: 01:41:38 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-0: 01:41:49 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:41:49 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-test-app-0: 01:41:49 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:41:51 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:41:51 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-another-database-0: 01:41:51 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-another-database-1: 01:41:52 ERROR unit.another-database/1.juju-log An exception occurred when installing charmed-postgresql. Reason: Snap: 'charmed-postgresql'; command ['snap', 'install', 'charmed-postgresql', '--channel="14/stable"', '--revision="124"'] failed with output = ''
machine-1: 01:41:56 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:41:56 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-test-app-1: 01:41:56 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-another-database-1: 01:42:03 ERROR unit.another-database/1.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-another-database-1/charm/./src/charm.py", line 1879, in <module>
    main(PostgresqlOperatorCharm)
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/main.py", line 553, in main
    manager.run()
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/main.py", line 529, in run
    self._emit()
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/main.py", line 518, in _emit
    _emit_charm_event(self.charm, self.dispatcher.event_name, self._juju_context)
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/main.py", line 139, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/framework.py", line 347, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/framework.py", line 853, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-another-database-1/charm/venv/ops/framework.py", line 943, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-another-database-1/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 735, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-another-database-1/charm/./src/charm.py", line 925, in _on_leader_elected
    self.update_config()
  File "/var/lib/juju/agents/unit-another-database-1/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 735, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-another-database-1/charm/./src/charm.py", line 1634, in update_config
    self._patroni.render_patroni_yml_file(
  File "/var/lib/juju/agents/unit-another-database-1/charm/lib/charms/tempo_k8s/v1/charm_tracing.py", line 735, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-another-database-1/charm/src/cluster.py", line 606, in render_patroni_yml_file
    version=self.get_postgresql_version().split(".")[0],
AttributeError: 'NoneType' object has no attribute 'split'

[32mINFO    [0m pytest_operator.plugin:plugin.py:991 Forgetting model main...